### PR TITLE
Add sample: Front Door Standard/Premium (Preview) with Web Application Firewall and custom rule set

### PR DESCRIPTION
**Pull request by johndowns**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add sample: Front Door Standard/Premium (Preview) with Web Application Firewall and custom rule set

### CODE REVIEW

File: 201-front-door-standard-premium-waf-custom/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/201-front-door-standard-premium-waf-custom/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-standard-premium-waf-custom%2Fazuredeploy.json)  [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-standard-premium-waf-custom%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-front-door-standard-premium-waf-custom%2Fazuredeploy.json)
```
Comment (bmoore-msft) [inline 201-front-door-standard-premium-waf-custom/README.md:13]: remove extra visualize button

File: 201-front-door-standard-premium-waf-custom/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+The following resources are deployed as part of the solution:
+
+### Prerequisites
+- Azure Storage with a static website, which acts as a simulated origin in this sample.
+
+### Front Door Standard/Premium (Preview)
+- Front Door profile, endpoint, origin group, origin, and route to direct traffic to the Azure Storage static website.
+  - Note that you can use either the standard or premium Front Door SKU for this sample. Custom rules for the WAF are supported in either SKU (note that managed rule sets require the premium SKU though). By default, the standard SKU is used.
+- Front Door WAF policy with a custom rule blocking requests from a defined set of IP address ranges.
+  - In this sample, the IP address 198.51.100.100 and the range 203.0.113.0/24 are both blocked. THese are within the [IANA IP address ranges reserved for documentation](https://tools.ietf.org/html/rfc5737).
```
Comment (bmoore-msft) [inline 201-front-door-standard-premium-waf-custom/README.md:30]: type in capitalization "THese"

File: 201-front-door-standard-premium-waf-custom/main.bicep [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+param frontDoorSkuName string = 'Standard_AzureFrontDoor'
+
+@allowed([
+  'Detection'
+  'Prevention'
+])
+@description('The mode that the WAF should be deployed using. In \'Prevention\' mode, the WAF will block requests it detects as malicious. In \'Detection\' mode, the WAF will not block requests and will simply log the request.')
```
Comment (bmoore-msft) [inline 201-front-door-standard-premium-waf-custom/main.bicep:21]: move this above @allowed

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-front-door-standard-premium-waf-custom/main.bicep (hunk 3 above)
Comment (johndowns) [reply 201-front-door-standard-premium-waf-custom/main.bicep]: Fixed. Is there a standard order of decorators?

File: 201-front-door-standard-premium-waf-custom/README.md (hunk 2 above)
Comment (johndowns) [reply 201-front-door-standard-premium-waf-custom/README.md]: Fixed

File: 201-front-door-standard-premium-waf-custom/README.md (hunk 1 above)
Comment (johndowns) [reply 201-front-door-standard-premium-waf-custom/README.md]: Fixed

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by azure-quickstarts** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: 201-front-door-standard-premium-waf-custom/main.bicep (hunk 3 above)
Comment (bmoore-msft) [reply 201-front-door-standard-premium-waf-custom/main.bicep]: I think you established one...

**Pull request merged**